iceberg/src/main/scala/org/apache/spark/sql/delta/serverSidePlanning/SparkToIcebergExpressionConverter.scala

```diff
@@ -0,0 +1,181 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.delta.serverSidePlanning
+
+import org.apache.spark.sql.sources._
+import org.apache.spark.sql.types.StructType
+import shadedForDelta.org.apache.iceberg.expressions.{Expression, Expressions}
+
+/**
+ * Converts Spark Filter expressions to Iceberg Expression objects for server-side planning.
+ *
+ * This is a pure utility class with no side effects. All methods are thread-safe and stateless.
+ *
+ * Filter Mapping Table:
+ * {{{
+ * +-----------------------+--------------------------------+
+ * | Spark Filter          | Iceberg Expression             |
+ * +-----------------------+--------------------------------+
+ * | EqualTo               | Expressions.equal()            |
+ * | LessThan              | Expressions.lessThan()         |
+ * | GreaterThan           | Expressions.greaterThan()      |
+ * | LessThanOrEqual       | Expressions.lessThanOrEqual()  |
+ * | GreaterThanOrEqual    | Expressions.greaterThanOrEqual()|
+ * | IsNull                | Expressions.isNull()           |
+ * | IsNotNull             | Expressions.notNull()          |
+ * | And                   | Expressions.and()              |
+ * | Or                    | Expressions.or()               |
+ * +-----------------------+--------------------------------+
+ * }}}
+ *
+ * Unsupported filters return None and are not pushed to the server.
+ * This only affects performance (server returns more data), not correctness
+ * (Spark always re-applies all filters locally as residuals).
+ *
+ * Example usage:
+ * {{{
+ *   val sparkFilter = EqualTo("id", 5)
+ *   val schema = StructType(...)
+ *   SparkToIcebergExpressionConverter.convert(sparkFilter, schema) match {
+ *     case Some(icebergExpr) => // Use expression
+ *     case None => // Filter not supported
+ *   }
+ * }}}
+ */
+object SparkToIcebergExpressionConverter {
+
+  /**
+   * Convert a Spark Filter to an Iceberg Expression.
+   *
+   * @param filter The Spark filter to convert
+   * @param schema The table schema (currently unused but reserved for future type conversions)
+   * @return Some(Expression) if the filter is supported, None otherwise
+   */
+  def convert(filter: Filter, schema: StructType): Option[Expression] = filter match {
+    // Equality and Comparison Operators
+    case EqualTo(attribute, value) =>
+      Some(convertEqualTo(attribute, value))              // Expressions.equal()
+    case LessThan(attribute, value) =>
+      Some(convertLessThan(attribute, value))             // Expressions.lessThan()
```

Comment:
i dont think the comments on the side are needed. they dont contribute any new info that not already obvious from the `case ****` code.